rpc/cc/arduino/cli/commands/v1/commands.proto

```diff
@@ -189,6 +189,8 @@ message InitRequest {
   string profile = 2;
   // The path where the sketch is stored
   string sketch_path = 3;
+  // Specificies if the default profile should be used
+  bool default_profile = 4;
```

Comment:
A better API is something like this:


```proto
message InitRequest {
  oneof message {
    NoProfile noProfile = 1;
    Profile profile = 2;
    DefaultProfile defaultProfile = 3;
  }
}

message NoProfile {
  Instance instance = 1;
}

message Profile {
  Instance instance = 1;
  string profile = 2;
}

message DefaultProfile {
 Instance instance = 1;
}
```

I left the `sketch_path` out on purpose. If somebody can explain how to use it, I can update the proposed API 👆

Reply:
> I left the sketch_path out on purpose. If somebody can explain how to use it, I can update the proposed API point_up_2

The profiles are written in a file `sketch.yaml` inside the sketch.

Having just the profile name it's not sufficient without knowing the path to the `sketch.yaml` it refers to, that's why the `InitReq` has the `sketch_path`. Basically to initialize a profile you need the pair `sketch_path` (where to find the `sketch.yaml`) and `profile` name.

Maybe `LoadSketch` could be a better place to retrieve the `sketch.yaml` content (at least the available profiles and the default one).

Also, now that you pointed out this API is lacking, I'm wondering how the CLI is obtaining the profiles in the first place, this smells like we are bypassing the gRPC API.

Reply:
> that's why the `InitReq` has the `sketch_path`

Thank you! It makes sense. I was unaware that the `sketch_path` has been introduced with `profile`.

> Maybe `LoadSketch` could be a better place to retrieve the `sketch.yaml` content (at least the available profiles and the default one).

Great idea. I hope to see this in the future. Please note that the `LoadSketch` requires an `ArduinoCoreService`, so the profile info cannot be there as it is required to initialize the `ArduinoCoreService`. For me, loading the profile info is closer to the `SettingsService` and its client.

Reply:
> Please note that the LoadSketch requires an ArduinoCoreService, so the profile info cannot be there as it is required to initialize the ArduinoCoreService

Good point, AFAICS `LoadSketch` do not use the `ArduinoCoreService` internally, I guess this field could be safely removed from `LoadSketch` (and maybe also from other sketch-related commands).

I guess this "feature" request is leading to a bigger breaking change in the gRPC API.

Reply:
> I guess this field could be safely removed from `LoadSketch` (and maybe also from other sketch-related commands).

💯 

> I guess this "feature" request is leading to a bigger breaking change in the gRPC API.

Please go ahead with this enhancement. I do not want to hold back the PR because of the gRPC API only.

However, the current gRPC API can be improved as it is not self-explanatory. As I wrote in https://github.com/arduino/arduino-cli/pull/2203#discussion_r1229459967, the only way to clarify how the `profile` and `default_profile` parameters connected is either with an API comment or a runtime error.

> What should happen when the profile file contains `a` and `b` profiles, the `default_profile` is `b` in the profile file, and the `profile` prop is `a` and `default_profile` is `true` of the `InitRequest`.

And if you think about it, `profile` and `default_profile` are not connected. You either use the default profile or you specify one.